sui_core/src/authority.rs

```diff
@@ -517,12 +518,38 @@ impl AuthorityState {
             return Err(SuiError::TooManyItemsError(MAX_ITEMS_LIMIT));
         }
 
-        let (batches, transactions) = self
+        let (mut batches, transactions): (
+            Vec<SignedBatch>,
+            Vec<(TxSequenceNumber, TransactionDigest)>,
+        ) = self
             ._database
             .batches_and_transactions(request.start, request.end)?;
 
+        // Append the transaction info to individual transactions if requested
+        let mut transactions_with_info = Vec::new();
+        for (seq, digest) in transactions.iter() {
+            if request.include_tx_info {
+                let tx_info = self.make_transaction_info(digest).await?;
```

Comment:
Two thoughts here:
1. Calling make_transactions_info for each TX is going to be very slow because each call is async.   What we really need is a call to the backend store with a range of sequence numbers, or a set of TX's.  I have other thoughts on refactoring the entire batching and transaction storage system, but now's not the time. :)
2. Is there a way we can avoid boxing tx_info in the line below?  It is expensive both in terms of heap allocation and runtime cost.